database/secret/create.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+//nolint:dupl // ignore similar code with update.go
+package secret
+
+import (
+	"fmt"
+
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// CreateSecret creates a new secret in the database.
+func (e *engine) CreateSecret(s *library.Secret) error {
+	// handle the secret based off the type
+	switch s.GetType() {
+	case constants.SecretShared:
+		e.logger.WithFields(logrus.Fields{
+			"org":    s.GetOrg(),
+			"team":   s.GetTeam(),
+			"secret": s.GetName(),
+			"type":   s.GetType(),
+		}).Tracef("creating secret %s/%s/%s/%s in the database", s.GetType(), s.GetOrg(), s.GetTeam(), s.GetName())
+	default:
+		e.logger.WithFields(logrus.Fields{
+			"org":    s.GetOrg(),
+			"repo":   s.GetRepo(),
+			"secret": s.GetName(),
+			"type":   s.GetType(),
+		}).Tracef("creating secret %s/%s/%s/%s in the database", s.GetType(), s.GetOrg(), s.GetRepo(), s.GetName())
+	}
```

Comment:
It looks like this handles Team(Shared) and Repo secrets here and in the error message below. What about org secrets? There isn't a repo name for org secrets.

Could you add test cases for each of the secret types?